src/diffusers/pipelines/stable_audio/pipeline_stable_audio.py

```diff
@@ -0,0 +1,863 @@
+from .modeling_stable_audio import StableAudioDiTModel, StableAudioProjectionModel
+
+
+if is_librosa_available():
+    pass
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+EXAMPLE_DOC_STRING = """
+    Examples:
+        ```py
+        >>> import scipy
+        >>> import torch
+        >>> from diffusers import StableAudioPipeline
+
+        >>> repo_id = "ylacombe/stable-audio-1.0"  # TODO (YL): change once set
+        >>> pipe = StableAudioPipeline.from_pretrained(repo_id, torch_dtype=torch.float16)
+        >>> pipe = pipe.to("cuda")
+
+        >>> # define the prompts
+        >>> prompt = "The sound of a hammer hitting a wooden surface."
+        >>> negative_prompt = "Low quality."
+
+        >>> # set the seed for generator
+        >>> generator = torch.Generator("cuda").manual_seed(0)
+
+        >>> # run the generation
+        >>> audio = pipe(
+        ...     prompt,
+        ...     negative_prompt=negative_prompt,
+        ...     num_inference_steps=200,
+        ...     audio_end_in_s=10.0,
+        ...     num_waveforms_per_prompt=3,
+        ...     generator=generator,
+        ... ).audios
+
+        >>> # save the best audio sample (index 0) as a .wav file
+        >>> scipy.io.wavfile.write("techno.wav", rate=16000, data=audio[0])
+        ```
+"""
+
+
+class StableAudioPipeline(DiffusionPipeline):
+    r"""
+    Pipeline for text-to-audio generation using StableAudio.
+
+    This model inherits from [`DiffusionPipeline`]. Check the superclass documentation for the generic methods
+    implemented for all pipelines (downloading, saving, running on a particular device, etc.).
+
+    Args:
+        vae ([`AutoencoderOobleck`]):
+            Variational Auto-Encoder (VAE) model to encode and decode images to and from latent representations.
+        text_encoder ([`~transformers.T5EncoderModel`]):
+            Frozen text-encoder. StableAudio uses the encoder of
+            [T5](https://huggingface.co/docs/transformers/model_doc/t5#transformers.T5EncoderModel), specifically the
+            [google-t5/t5-base](https://huggingface.co/google-t5/t5-base) variant.
+        projection_model ([`StableAudioProjectionModel`]):
+            A trained model used to linearly project the hidden-states from the text encoder model and the start and
+            end seconds. The projected hidden-states from the encoder and the conditional seconds are concatenated to
+            give the input to the transformer model.
+        tokenizer ([`~transformers.T5Tokenizer`]):
+            Tokenizer to tokenize text for the frozen text-encoder.
+        transformer ([`StableAudioDiTModel`]):
+            A `StableAudioDiTModel` to denoise the encoded audio latents.
+        scheduler ([`EDMDPMSolverMultistepScheduler`]):
+            A scheduler to be used in combination with `transformer` to denoise the encoded audio latents.
+    """
+
+    model_cpu_offload_seq = "text_encoder->projection_model->transformer->vae"
```

Comment:
The cpu offload doesn't seem to work! 
Before diving into it on my side, do you think of any obvious reasons for the faliing tests ?

```py
FAILED tests/pipelines/stable_audio/test_stable_audio.py::StableAudioPipelineFastTests::test_cpu_offload_forward_pass_twice - AssertionError: False is not true : Not offloaded: ['vae']
FAILED tests/pipelines/stable_audio/test_stable_audio.py::StableAudioPipelineFastTests::test_model_cpu_offload_forward_pass - AssertionError: False is not true : Not offloaded: ['vae']
FAILED tests/pipelines/stable_audio/test_stable_audio.py::StableAudioPipelineFastTests::test_sequential_cpu_offload_forward_pass - NotImplementedError: Cannot copy out of meta tensor; no data!
FAILED tests/pipelines/stable_audio/test_stable_audio.py::StableAudioPipelineFastTests::test_sequential_offload_forward_pass_twice - NotImplementedError: Cannot copy out of meta tensor; no data!
```

Reply:
Yes, it's because we call `vae.decode` and not the forward pass

Reply:
Yes, it's because we call vae.decode and not the forward pass